Backend authentication

Unicus-Tom opened the pull request.

Unicus-Tom committed:
Added JSON token support

Currently accepts json input with fields "username, idToken, Expiration" content of fields are not yet used for anything and returns a json token.

Unicus-Tom committed:
Added CORS

Included CORS configuration as this was needed for front end and back end to work together properly.
Defaults to allowing all domains. Support for putting domains in config coming later.

Co-Authored-By: kris2ffer <kris2ffer@users.noreply.github.com>

Unicus-Tom committed:
Added config support

Added support for json based config files.

Unicus-Tom committed:
Added /refreshtoken to api

You can now use an old sepestoken to get a new token.

gjersvik commented on RestApi/Sepes.RestApi/appsettings.Development.json:
```diff
@@ -1,9 +1,22 @@
+    "Issuer": "example.com"
+  },
+  "AzureLogToken": {
+    "ServiceApiKey": "AZURETokenHere"
```
You should not need to have any config values here that are the same as in appsettings.json

gjersvik commented on RestApi/Sepes.RestApi/Startup.cs:
```diff
@@ -30,10 +42,44 @@ public void ConfigureServices(IServiceCollection services)
+            {  
+            options.TokenValidationParameters = new TokenValidationParameters  
+                {  
+                ValidateIssuer = false,  //TODO set to true before MVP
```
Small tings like this that are easy to forget should realy be GitHub issues.

Unicus-Tom replied:
Sure, I will get it done before my last commit

gjersvik reviewed (approved):
I would recommenced turning some of these todos into GitHub issues.

Kris2ffer commented on RestApi/Sepes.RestApi/Controllers/AuthController.cs:
```diff
@@ -13,6 +23,102 @@ namespace Sepes.RestApi.Controller
+        public string Username { get; set; }
+        public string idToken { get; set; }
+        public string Expiration { get; set; }
     }
```
Would like to see one generic token class instead of two that are basically the same.

Unicus-Tom replied:
They represent two different token formats. Currently contain the same content but may differ in the future. If it turns out we dont need two different classes it is trivial to combine them later.

Unicus-Tom committed:
Converted TODOs into issues

Trimmed down comments made for my own sake and completed my TODOs or made them into issues

Unicus-Tom committed:
Merge branch 'master' into backend-authentication
